tests/contrib/molten/test_molten.py

```diff
@@ -415,3 +426,82 @@ def test_http_request_header_tracing(self):
         span = spans[0]
         self.assertEqual(span.name, "molten.request")
         self.assertEqual(span.get_tag("http.request.headers.my-header"), "my_value")
+
+    def test_inferred_spans_api_gateway_default(self):
+        headers = {
+            "x-dd-proxy": "aws-apigateway",
+            "x-dd-proxy-request-time-ms": "1736973768000",
+            "x-dd-proxy-path": "/",
+            "x-dd-proxy-httpmethod": "GET",
+            "x-dd-proxy-domain-name": "local",
+            "x-dd-proxy-stage": "stage",
+        }
+
+        distributed_headers = {
+            "x-dd-proxy": "aws-apigateway",
+            "x-dd-proxy-request-time-ms": "1736973768000",
+            "x-dd-proxy-path": "/",
+            "x-dd-proxy-httpmethod": "GET",
+            "x-dd-proxy-domain-name": "local",
+            "x-dd-proxy-stage": "stage",
+            "x-datadog-trace-id": "1",
+            "x-datadog-parent-id": "2",
+            "x-datadog-origin": "rum",
+            "x-datadog-sampling-priority": "2",
+        }
+
+        for setting_enabled in [False, True]:
```

Comment:
Why not rely on parametrize like how we did for the other tests?

Reply:
I did try but kept getting errors about positional arguments when using `@pytest.mark.parametrize`
>  TypeError: TestMolten.test_inferred_spans_api_gateway_default() missing 1 required positional argument

The problem is that TestMolten is set up to use TracerTestCase, which is a `unittest.TestCase` which cannot use parametrization per their docs: https://docs.pytest.org/en/stable/how-to/unittest.html#pytest-features-in-unittest-testcase-subclasses

I found this out after stumbling into this Stack Overflow thread: https://stackoverflow.com/questions/63720118/pytest-parametrize-i-am-getting-missing-required-positional-arguments

There are some workarounds, but I didn't want to refactor all the Molten tests.